subprojects/gradle/instrumentation-tests/src/integTest/kotlin/com/avito/instrumentation/reservation/client/kubernetes/KubernetesReservationClientIntegrationTest.kt

```diff
@@ -0,0 +1,173 @@
+package com.avito.instrumentation.reservation.client.kubernetes
+
+import com.avito.instrumentation.reservation.adb.AndroidDebugBridge
+import com.avito.instrumentation.reservation.adb.EmulatorsLogsReporter
+import com.avito.instrumentation.reservation.client.ReservationClient
+import com.avito.instrumentation.reservation.request.Device.CloudEmulator
+import com.avito.instrumentation.reservation.request.Reservation
+import com.avito.instrumentation.reservation.request.createStubInstance
+import com.avito.runner.service.worker.device.adb.Adb
+import com.avito.truth.assertThat
+import com.avito.utils.gradle.KubernetesCredentials
+import com.avito.utils.gradle.createKubernetesClient
+import com.avito.utils.logging.FakeCILogger
+import com.google.common.truth.Truth.assertThat
+import io.fabric8.kubernetes.client.KubernetesClientException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import java.io.File
+import java.net.UnknownHostException
+
```

Comment:
What about [`KubernetesServer`](https://github.com/fabric8io/kubernetes-client#mocking-kubernetes) mocks?
Will it be helpful?

Reply:
If i understand correctly, you're proposing to record k8s responses and convert this tests to hermetic?
i am hands down like this approach, but want to keep flag to switch between integration and mock/stub variant;
I see a value here in testing migration on newer k8s versions, that could behave differently

Reply:
I didn't mean a total replacement in this PR, just wondering about plans on this topic.